test/Microsoft.ML.TestFrameworkCommon/Attributes/MLNETFactAttribute.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.ML.TestFrameworkCommon.Attributes
+{
+    /// <summary>
+    /// ML.NET facts that will retry several flaky test cases, use default timeout settings
+    /// </summary>
+    [XunitTestCaseDiscoverer("Microsoft.ML.TestFrameworkCommon.MLNETFactDiscoverer", "Microsoft.ML.TestFrameworkCommon")]
+    public class MLNETFactAttribute : FactAttribute
+    {
+        /// <summary>
+        /// Number of retries allowed for a failed test. If unset (or set less than 1), will
+        /// default to 3 attempts.
```

Comment:
Two issues here:

1. If this attribute is used for all tests, it needs to default to 1
2. If this attribute is used only for tests known to be flaky, the maximum acceptable default is 2

Reply:
We intend to retry all failed test cases and log all the failing for investigate. All retry is to unblock every one from passing their PR build for now as failing is random and frequent and investigating flaky test can be time consuming. 

---
In reply to: [354508994](https://github.com/dotnet/machinelearning/pull/4510#discussion_r354508994) [](ancestors = 354508994)